deployment/environment/memory/zksync_chains.go

```diff
@@ -0,0 +1,85 @@
+package memory
+
+import (
+	"context"
+	"math/big"
+	"strconv"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/hashicorp/consul/sdk/freeport"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/zksync-sdk/zksync2-go/accounts"
+	"github.com/zksync-sdk/zksync2-go/clients"
+
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/framework/components/blockchain"
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+func GenerateChainsZk(t *testing.T, numChains int) map[uint64]deployment.Chain {
+	chains := make(map[uint64]deployment.Chain)
+
+	for i := 0; i < numChains; i++ {
+		chainID := chain_selectors.TEST_90000051.EvmChainID + uint64(i) //nolint:gosec // it shouldn't overflow
+
+		output, err := blockchain.NewBlockchainNetwork(&blockchain.Input{
+			Type:    "anvil-zksync",
+			ChainID: strconv.FormatUint(chainID, 10),
+			Port:    strconv.FormatInt(int64(freeport.GetN(t, 1)[0]), 10),
+		})
+		require.NoError(t, err)
+
+		testcontainers.CleanupContainer(t, output.Container)
+
+		sel, err := chain_selectors.SelectorFromChainId(chainID)
+		require.NoError(t, err)
+
+		client, err := ethclient.Dial(output.Nodes[0].ExternalHTTPUrl)
+		require.NoError(t, err)
+
+		gasPrice, err := client.SuggestGasPrice(context.Background())
+		require.NoError(t, err)
+
+		keyedTransactors := make([]*bind.TransactOpts, 0)
+		for _, pk := range blockchain.AnvilZKSyncRichAccountPks {
+			privateKey, err := crypto.HexToECDSA(pk)
+			require.NoError(t, err)
+			transactor, err := bind.NewKeyedTransactorWithChainID(privateKey, new(big.Int).SetUint64(chainID))
+			transactor.GasPrice = gasPrice
+			require.NoError(t, err)
+			keyedTransactors = append(keyedTransactors, transactor)
+		}
+
+		clientZk := clients.NewClient(client.Client())
+		deployerZk, err := accounts.NewWallet(common.Hex2Bytes(blockchain.AnvilZKSyncRichAccountPks[0]), clientZk, nil)
```

Comment:
some check to ensure AnvilZKSyncRichAccountPks length is as expected

Reply:
Added checks d1ec77c2de35c1d89ec78bc5ad56fa7f5b685672